src/main/kotlin/eu/europa/ec/eudi/pidissuer/adapter/out/jose/JWKExtensions.kt

```diff
@@ -22,9 +22,8 @@ import com.nimbusds.jose.jwk.JWK
 /**
  * Converts this [JWK] to an [ECKey] or raises [onFailure].
  */
-context(Raise<E>)
-internal fun <E> JWK.toECKeyOrFail(onFailure: () -> E): ECKey =
-    when (this) {
-        is ECKey -> this
+internal fun <E> Raise<E>.toECKeyOrFail(jwk: JWK, onFailure: () -> E): ECKey =
```

Comment:
@babisRoutis I think the method would feel more 'organic' if it was defined as `JWK.toECKeyOrFail(onFailure: () -> E): Either<E, ECKey>`

Reply:
Reworked `JwkExtensions` as follows:

Introduced a new interface 
```korlin
internal interface JWKExtensions<in Error> : Raise<Error> {
    /**
     * Converts this [JWK] to an [ECKey] or raises [onFailure].
     */
    fun JWK.toECKeyOrFail(onFailure: () -> Error): ECKey =
        when (this) {
            is ECKey -> this
            else -> raise(onFailure())
        }
}
```
This interface has the `toECKeyOrFail` method having the `JWK` as receiver parameter, which  seems more natural choice.
In addition, a new method was added that "lifts" the above interface/method in the context of a `Raise<E>` 

```kotlin
internal fun <Error> Raise<Error>.jwkExtensions(): JWKExtensions<Error> =
    object : JWKExtensions<Error>, Raise<Error> by this {}
```